pysat/_instrument.py

```diff
@@ -675,7 +675,8 @@ def _index(self, data=None):
             if 'time' in data.indexes:
                 return data.indexes['time']
             elif 'Epoch' in data.indexes:
-                return data.indexes['Epoch']
+                return pds.to_datetime([dt.datetime.utcfromtimestamp(x/1000)
```

Comment:
Thanks! This would be better in the load routine though. Epoch should be in dates and times, not a super large float.

Reply:
or at least in the icon instrument files. Different instruments have different timings which isn't really supported in the load method at the moment.